### PR TITLE
Removed unnecessary copy of Attributes struct in mustache rendering

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

File: cf-agent/verify_files.c [hunk 1]
```diff
@@ -624,20 +624,19 @@ static bool SaveBufferCallback(const char *dest_filename, void *param, NewLineMo
     return true;
 }
 
-
-static PromiseResult RenderTemplateMustache(EvalContext *ctx, const Promise *pp, const Attributes *attr,
+static PromiseResult RenderTemplateMustache(EvalContext *ctx, const Promise *pp, const Attributes *a,
```
Comment (lluiscampos) [inline cf-agent/verify_files.c:627]: I would keep `attr` to be consistent with most of the other functions signatures

File: cf-agent/verify_files.c [hunk 2]
```diff
@@ -624,20 +624,19 @@ static bool SaveBufferCallback(const char *dest_filename, void *param, NewLineMo
...
     JsonElement *destroy_this = NULL;
 
-    if (a.template_data == NULL)
+    if (template_data == NULL)
     {
-        a.template_data = DefaultTemplateData(ctx, NULL);
-        destroy_this = a.template_data;
+        destroy_this = DefaultTemplateData(ctx, NULL);
+        template_data = destroy_this;
     }
```
Comment (lluiscampos) [inline cf-agent/verify_files.c:640]: This makes sense to me :+1:

**Comment by olehermanse** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/5278/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/5278/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-5278/

File: cf-agent/verify_files.c (hunk 1 above)
Comment (olehermanse) [reply cf-agent/verify_files.c]: @lluiscampos Thanks, I agree, fixed.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/5283/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/5283/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-5283/

File: cf-agent/verify_files.c [hunk 3]
```diff
@@ -650,33 +649,33 @@ static PromiseResult RenderTemplateMustache(EvalContext *ctx, const Promise *pp,
     Buffer *output_buffer = BufferNew();
 
     char *message;
-    if ( strcmp("inline_mustache", a.template_method) == 0)
+    if ( strcmp("inline_mustache", attr->template_method) == 0)
```
Comment (vpodzime) [inline cf-agent/verify_files.c:652]: Might as well fix the extra space before `strcmp` here.

Comment (olehermanse) [reply cf-agent/verify_files.c:652]: But then I might as well change it to `StringEqual()`, and then:

MindlessBelovedAmericanlobster-size_restricted

Comment (olehermanse) [reply cf-agent/verify_files.c]: Removed the space.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/5285/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/5285/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-5285/

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/5288/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/5288/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-5288/

**Pull request merged**